### ISSUE TITLE
QThread: Destroyed while thread is still running

### ISSUE THREAD

**Issue by marioortizmanero** [vidify/vidify#54]
This error appears sometimes when using audiosync and crashes the app. It's probably related to starting a new audiosync thread and having to kill the previous one without waiting for them to finish.

**Comment by marioortizmanero** [comment]
Possibly related bug when quickly switching between songs:

```
QThread::wait: Thread tried to wait on itself
zsh: segmentation fault (core dumped)  python3 -m vidify --debug
```

**Comment by marioortizmanero** [comment]
Maybe [QThreadPool](https://doc.qt.io/qtforpython/PySide2/QtCore/QThreadPool.html) would be a better fit for this usage.

**Comment by marioortizmanero** [comment]
Another thing to consider is rather than inheriting from QThread, declaring the worker as a QObject and then move the worker to a new thread. After reading the docs, I've come to the conclusion that [for the Audiosync loop it doesn't make sense to do this](https://github.com/vidify/vidify/blob/master/vidify/audiosync.py#L6). But for the YoutubeDL metadata download, it should actually work this way.

**Comment by marioortizmanero** [comment]
This was finally fixed with the last v2.1 update. It only happens sometimes when the app is forcefully closed.
